server/tests/chart_test.py

```diff
@@ -90,84 +90,78 @@ def test_cache_necessary_dates(self, mock_stat_vars, mock_landing_page):
         }
         with app.app_context():
             app.config['CHART_CONFIG'] = [{
-                'label':
-                    'Test',
-                'charts': [{
-                    'title':
-                        'Test1',
-                    "chartType":
-                        "GROUP_BAR",
-                    "axis":
-                        "PLACE",
-                    "statsVars": [
-                        "StatVar1",
-                        "StatVar2",
-                        "StatVar3",
-                        "StatVar4",
-                        "StatVar5",
-                    ]
-                }],
-                'children': [{
-                    'label':
-                        'Test',
-                    'charts': [{
-                        'title':
-                            'Test2',
-                        "chartType":
-                            "SINGLE_BAR",
-                        "statsVars": [
-                            "StatVar1",
-                            "StatVar2",
-                            "StatVar6",
-                            "StatVar7",
-                            "StatVar8",
-                        ]
-                    }]
-                }]
+                "category": ["Test", "Test1"],
+                "title": "Test1",
+                "statsVars": [
+                    "StatVar1", "StatVar2", "StatVar3", "StatVar4", "StatVar5"
+                ],
+                "isOverview": True
+            }, {
+                "category": ["Test", "Test2"],
+                "title":
+                    "Test2",
+                "statsVars": [
+                    "StatVar1", "StatVar2", "StatVar6", "StatVar7", "StatVar8"
+                ]
             }]
-            response = app.test_client().get('api/chart/config/geoId/06')
+            response = app.test_client().get('api/chart/data/geoId/06')
             assert response.status_code == 200
             assert json.loads(response.data)['data'] == {
                 'geoId/06': {
                     'StatVar1': {
                         'data': {
+                            '2010': 200,
```

Comment:
@beets I want to confirm the original test intends to omit year 2010 as this is used for bar charts with latest date?

Reply:
yea, we were omitting data if the stat var was only used in bar charts. initially only keeping the latest date. then this was fixed in #389, where we are trying to get the best latest date to keep (so that the same latest date is chosen across stat vars). 

@chejennifer

Reply:
Got it. Now primary charts are line charts with all years data. So the test update seems ok.